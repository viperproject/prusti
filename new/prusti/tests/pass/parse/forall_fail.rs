// compile-flags: -Zprint-desugared-specs -Zprint-typeckd-specs -Zskip-verify -Zhide-uuids
// normalize-stdout-test: "[a-z0-9]{32}" -> "$(NUM_UUID)"
// normalize-stdout-test: "[a-z0-9]{8}-[a-z0-9]{4}-[a-z0-9]{4}-[a-z0-9]{4}-[a-z0-9]{12}" -> "$(UUID)"

#![feature(register_tool)]
#![register_tool(prusti)]

use prusti_contracts::*;

#[requires(forall)]
fn test1() {}

#[requires(forall())]
fn test2() {}

#[requires(forall(|))]
fn test3() {}

#[requires(forall(||) 1+1)]
fn test4() {}

#[requires(forall(|a, b| true))]
fn test5() {}

#[requires(forall(||) || forall(||))]
fn test6() {}

#[requires(forall(|| 1+1 == 1+1, triggers=[1]))]
fn test7() {}

#[requires(forall(|| true, triggers=[(1,2), 1]))]
fn test8() {}

#[requires(forall(|| true, triggers=1))]
fn test9() {}

#[requires(forall(||))]
fn test10() {}

#[requires(forall(|| 1+1 == 1+1))]
fn test11() {}

#[requires(forall(||, triggers=[]))]
fn test12() {}

#[requires(forall(|| 1+1 == 1+1, triggers=[(1,)]))]
fn test13() {}

fn main() {}
