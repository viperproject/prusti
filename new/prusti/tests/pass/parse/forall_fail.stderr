error: expected `(`
  --> $DIR/forall_fail.rs:10:12
   |
10 | #[requires(forall)]
   |            ^^^^^^

error: expected `|`
  --> $DIR/forall_fail.rs:13:18
   |
13 | #[requires(forall())]
   |                  ^^

error: a quantifier must have at least one argument
  --> $DIR/forall_fail.rs:16:18
   |
16 | #[requires(forall(|))]
   |                  ^^^

error: a quantifier must have at least one argument
  --> $DIR/forall_fail.rs:19:18
   |
19 | #[requires(forall(||) 1+1)]
   |                  ^^^^

error: expected `:`
  --> $DIR/forall_fail.rs:22:21
   |
22 | #[requires(forall(|a, b| true))]
   |                     ^

error: a quantifier must have at least one argument
  --> $DIR/forall_fail.rs:25:18
   |
25 | #[requires(forall(||) || forall(||))]
   |                  ^^^^

error: a quantifier must have at least one argument
  --> $DIR/forall_fail.rs:28:18
   |
28 | #[requires(forall(|| 1+1 == 1+1, triggers=[1]))]
   |                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: a quantifier must have at least one argument
  --> $DIR/forall_fail.rs:31:18
   |
31 | #[requires(forall(|| true, triggers=[(1,2), 1]))]
   |                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: a quantifier must have at least one argument
  --> $DIR/forall_fail.rs:34:18
   |
34 | #[requires(forall(|| true, triggers=1))]
   |                  ^^^^^^^^^^^^^^^^^^^^^

error: a quantifier must have at least one argument
  --> $DIR/forall_fail.rs:37:18
   |
37 | #[requires(forall(||))]
   |                  ^^^^

error: a quantifier must have at least one argument
  --> $DIR/forall_fail.rs:40:18
   |
40 | #[requires(forall(|| 1+1 == 1+1))]
   |                  ^^^^^^^^^^^^^^^

error: a quantifier must have at least one argument
  --> $DIR/forall_fail.rs:43:18
   |
43 | #[requires(forall(||, triggers=[]))]
   |                  ^^^^^^^^^^^^^^^^^

error: a quantifier must have at least one argument
  --> $DIR/forall_fail.rs:46:18
   |
46 | #[requires(forall(|| 1+1 == 1+1, triggers=[(1,)]))]
   |                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: aborting due to 13 previous errors

